tests/test_kb_dataset.py

```diff
@@ -0,0 +1,127 @@
+import argparse
+import sys
+import unittest
+from collections import defaultdict
+from pathlib import Path
+
+import datasets
+
+
+
+class TestKBDataset(unittest.TestCase):
+    def setUp(self) -> None:
+        self.dataset_source = datasets.load_dataset(sys.argv[1], name="source")
+        self.dataset_bigbio = datasets.load_dataset(sys.argv[1], name="bigbio")
+
+    def print_statistics(self):
+        for split_name, split in self.dataset_bigbio.items():
+            print(split_name)
+            print("=" * 10)
+
+            counter = defaultdict(int)
+            for example in split:
+                for feature_name, feature in features.items():
+                    if isinstance(feature, datasets.Value) or isinstance(
+                        feature_name, dict
+                    ):
+                        if example[feature_name]:
+                            counter[feature_name] += 1
+                    else:
+                        counter[feature_name] += len(example[feature_name])
+
+            for k, v in counter.items():
+                print(f"{k}: {v}")
+            print()
+
+    def runTest(self):
+        self.print_statistics()
+        self.test_is_bigbio_schema_compatible()
+        self.test_are_ids_globally_unique()
+        self.test_do_all_referenced_ids_exist()
+
+    def test_is_bigbio_schema_compatible(self):
+        for split in self.dataset_bigbio.values():
+            self.assertEqual(split.features, features)
```

Comment:
My initial reaction was to not make all the features required, but I can see the simplicity of things like this if we do make everything required. I'm on the fence, but probably still slightly leaning towards not making all fields required.